src/tests/suite/daos_dist_tx.c

```diff
@@ -377,11 +377,15 @@ dtx_10(void **state)
 	req.arg->expect_result = -DER_NONEXIST;
 	punch_dkey_with_flags(dkey2, th, &req, DAOS_COND_PUNCH);
 
-	req.arg->expect_result = 0;
-	punch_dkey_with_flags(dkey, th, &req, DAOS_COND_PUNCH);
-
+	/** Remove the test for the dkey because it can't work with client
+	 *  side caching and punch propagation.   The dkey will have been
+	 *  removed by the akey punch above.  The problem is the server
```

Comment:
When the akey punch leaves the tree empty, it punches the dkey.  That's the punch propagation algorithm.   The issue is it's a conditional punch.   In the absence of caching, the dkey punch would fail with -DER_NONEXIST because it would have already been punched.   With caching, it fails on commit with -DER_NONEXIST because it's already been punched by the time the dkey punch executes.

Reply:
So that means if all the child name entries under the directory are punched (removed), then such directory will be automatically removed when punch the last child name entry? Do we allow an empty directory to exist?